src/tests/Services/CollectCmsDataServiceTest.php

```diff
@@ -0,0 +1,156 @@
+        $this->security_helper_mock->expects($this->never())->method('validate_collect_data_signature');
+
+        $this->assertNull($this->collect_cms_data_service->handle_collect_cms_data());
+    }
+
+    public function test_handle_collect_cms_data_with_invalid_signature()
+    {
+        $_SERVER['HTTP_X_ALMA_SIGNATURE'] = 'invalid_signature';
+
+        $this->security_helper_mock->expects($this->once())
+            ->method('validate_collect_data_signature')
+            ->willThrowException(new AlmaInvalidSignatureException("Invalid signature"));
+
+        $this->alma_logger_mock->expects($this->once())
+            ->method('error')
+            ->with("Invalid signature");
+
+        $this->functions_proxy_mock->expects($this->once())
+            ->method('send_http_response')
+            ->with(array('error' => 'Invalid signature'), 403);
+
+        $this->payload_formatter_mock->expects($this->never())->method('formatConfigurationPayload');
+
+        $this->assertNull($this->collect_cms_data_service->handle_collect_cms_data());
+
+    }
+
+    public function test_handle_collect_cms_data_with_valid_signature()
+    {
+        $_SERVER['HTTP_X_ALMA_SIGNATURE'] = 'valid_signature';
+
+        $this->security_helper_mock->expects($this->once())
+            ->method('validate_collect_data_signature');
+
+        $cms_info = new CmsInfo([
+            'cms_name' => 'WooCommerce',
+            'cms_version' => 'v.4.2',
+            'third_parties_plugins' => [],
+            'theme_name' => 'Storefront',
+            'theme_version' => 'v.4.5',
+            'language_name' => 'PHP',
+            'language_version' => phpversion(),
+            'alma_plugin_version' => ALMA_VERSION,
+            'alma_sdk_version' => Client::VERSION,
+            'alma_sdk_name' => 'alma/alma-php-client',
+        ]);
+
+        $cms_features = new CmsFeatures([
+            'alma_enabled' => true,
+            'widget_cart_activated' => true,
+            'widget_product_activated' => false,
+            'used_fee_plans' => ['general_1_0_0' => ['enabled' => true, 'min_amount' => 0, 'max_amount' => 1000]],
+            'in_page_activated' => false,
+            'log_activated' => true,
+            'excluded_categories' => [],
+            'is_multisite' => is_multisite(),
+        ]);
+
+        $payload_formatter_return = [
+            'cms_info' => $cms_info->getProperties(),
+            'cms_features' => $cms_features->getProperties()
+        ];
+        $this->payload_formatter_mock->expects($this->once())
+            ->method('formatConfigurationPayload')
+            ->with($cms_info, $cms_features)
+            ->willReturn($payload_formatter_return);
+
+        $this->functions_proxy_mock->expects($this->once())
+            ->method('send_http_response')
+            ->with($payload_formatter_return, 200);
+
+        $this->assertNull($this->collect_cms_data_service->handle_collect_cms_data());
+    }
+}
```

Comment:
A new line is needed at each end of file

Reply:
Done